enterprise/frontend/src/embedding-sdk/cli/snippets/analytics-provider-snippet.ts

```diff
@@ -1,14 +1,28 @@
 export const ANALYTICS_PROVIDER_SNIPPET_MINIMAL = `
 import {createContext, useState} from 'react'
 
-// Used for the example theme switcher component
-export const AnalyticsContext = createContext({})
+/**
+ * @typedef {Object} AnalyticsContextType
+ * @property {'light'|'dark'} themeKey - The current theme key.
+ * @property {null} [email] - Email of the user.
+ * @property {(themeKey: 'light'|'dark') => void} setThemeKey - Function to update the theme.
+ */
+
+/**
+ * @type {React.Context<AnalyticsContextType>}
+ */
+export const AnalyticsContext = createContext(
+  /** @type {AnalyticsContextType} */ ({})
```

Comment:
I think this would be inferred by `type {React.Context<AnalyticsContextType>}` above, so I don't think we need this

Reply:
No, it's not

![image](https://github.com/user-attachments/assets/f374f41c-0bb0-49ab-b747-d740a25c1e3a)

Reply:
ah. can we not type `@type {React.Context<AnalyticsContextType>}` instead if we already have the inner `AnalyticsContextType` typed?

Reply:
This is news to me though, I didn't knew that JSDoc is that weak in terms of type inference 😢

Reply:
> can we not type @type {React.Context<AnalyticsContextType>} instead 

Checking

Reply:
Works, seems

Reply:
nice! let's do that then to trim the snippets a bit